### PR TITLE
Update tutorial for vm1.3.1

### PR DESCRIPTION
**Pull request by StanislavBreadless**

### CODE REVIEW

File: contracts/TwoUserMultisig.sol [hunk 1]
```diff
@@ -42,24 +42,21 @@ contract TwoUserMultisig is IAccount, IERC1271 {
         bytes32,
         bytes32 _suggestedSignedHash,
         Transaction calldata _transaction
-    ) external payable override onlyBootloader {
-        _validateTransaction(_suggestedSignedHash, _transaction);
+    ) external payable override onlyBootloader returns (bytes4 magic) {
```
Comment (vladbochok) [inline contracts/TwoUserMultisig.sol:45]: There are both declared `magic` and return statement. Please choose one option.

File: deploy/deploy-multisig.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ export default async function (hre: HardhatRuntimeEnvironment) {
     nonce: await provider.getTransactionCount(multisigAddress),
     type: 113,
     customData: {
-      ergsPerPubdata: utils.DEFAULT_ERGS_PER_PUBDATA_LIMIT,
+      ergsPerPubdata: utils.DEFAULT_GAS_PER_PUBDATA_LIMIT,
```
Comment (vladbochok) [inline deploy/deploy-multisig.ts]:
```suggestion
      gasPerPubdata: utils.DEFAULT_GAS_PER_PUBDATA_LIMIT,
```

File: contracts/TwoUserMultisig.sol [hunk 3]
```diff
@@ -72,10 +69,17 @@ contract TwoUserMultisig is IAccount, IERC1271 {
...
-        );
+        // The fact there is are enough balance for the account
+        // should be checked explicitly to prevent user paying for fee for a
+        // transaction that wouldn't be included on Ethereum.
+        uint256 totalRequiredBalance = _transaction.totalRequiredBalance();
+        require(totalRequiredBalance <= address(this).balance, "Not enough balance for fee + value");
+
+        if (isValidSignature(txHash, _transaction.signature) == EIP1271_SUCCESS_RETURN_VALUE) {
+            magic = ACCOUNT_VALIDATION_SUCCESS_MAGIC;
+        } else {
```
Comment (vladbochok) [inline contracts/TwoUserMultisig.sol]: else branch is redundant

File: contracts/TwoUserMultisig.sol [hunk 4]
```diff
@@ -121,27 +113,109 @@ contract TwoUserMultisig is IAccount, IERC1271 {
...
+
+        if (_signature.length != 130) {
+            // Signature is invalid anyway, but we need to proceed with the signature verification as usual
+            // in order for the fee estimation to work correctly
+            _signature = new bytes(130);
+            
+            // Making sure that the signatures look like a valid ECDSA signature and are not rejected rightaway
+            // while skipping the main verification process.
+            _signature[64] = bytes1(uint8(27));
+            _signature[129] = bytes1(uint8(27));
```
Comment (delaaxe) [inline contracts/TwoUserMultisig.sol:135]: this signature substitution logic happens when called from `validateTransaction`, but also during regular EIP1271  read calls.

is this intended?

**Comment by uF4No** [comment]
I'm getting an error when running the `deploy-multisig.ts` script in this tep:

```
  const tx = await aaFactory.deployAccount(
    salt,
    owner1.address,
    owner2.address,
    { gasLimit: 500000 } // added later
  );
  await tx.wait();
``` 

Without gasLimit, it fails with:
```
Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"error\":{\"code\":3,\"message\":\"Failed to submit transaction: cannot estimate transaction: out of gas.\",\"data\":{\"code\":104,\"message\":\"cannot estimate transaction: out of gas.\"}},\"id\":50}\n","error":{"code":3,"data":{"code":104,"message":"cannot estimate transaction: out of gas."}}
```


Setting `gasLimit: 500000` the transaction goes through but fails. For example https://goerli.explorer.zksync.io/tx/0xdd79071f9998794884596c6be76b21d0531c11c37e55af052c9bea40ae8600bb

File: contracts/TwoUserMultisig.sol (hunk 4 above)
Comment (StanislavBreadless) [reply contracts/TwoUserMultisig.sol:135]: @delaaxe it was just more convenient in this particular code, but no, it is not required. The substitution could be moved to other places

**Review by uF4No — APPROVED** [review]
Cloned, tested, all ok

**Pull request merged**